crates/primitives/starknet/src/execution/felt252_wrapper.rs

```diff
@@ -0,0 +1,404 @@
+        match FieldElement::from_byte_slice_be(bytes) {
+            Ok(ff) => Ok(Self(ff)),
+            Err(e) => match e {
+                FromByteSliceError::InvalidLength => Err(Felt252WrapperError::InvalidLength),
+                FromByteSliceError::OutOfRange => Err(Felt252WrapperError::OutOfRange),
+            },
+        }
+    }
+}
+
+/// [`Felt252Wrapper`] from u128.
+impl From<u128> for Felt252Wrapper {
+    fn from(value: u128) -> Self {
+        Felt252Wrapper::try_from(U256::from(value)).unwrap()
+    }
+}
+
+/// [`Felt252Wrapper`] to [`U256`].
+impl From<Felt252Wrapper> for U256 {
+    fn from(felt: Felt252Wrapper) -> Self {
+        U256::from_big_endian(&felt.0.to_bytes_be())
+    }
+}
+
+/// [`Felt252Wrapper`] from [`U256`].
+/// Overflow may occur and return [`Felt252WrapperError::OutOfRange`].
+impl TryFrom<U256> for Felt252Wrapper {
+    type Error = Felt252WrapperError;
+
+    fn try_from(u256: U256) -> Result<Self, Felt252WrapperError> {
+        let mut buf: [u8; 32] = [0; 32];
+        u256.to_big_endian(&mut buf);
+
+        Felt252Wrapper::try_from(&buf)
+    }
+}
+
+/// [`Felt252Wrapper`] from [`H256`].
+/// Overflow may occur and return [`Felt252WrapperError::OutOfRange`].
+impl TryFrom<H256> for Felt252Wrapper {
+    type Error = Felt252WrapperError;
+
+    fn try_from(h: H256) -> Result<Self, Felt252WrapperError> {
+        Felt252Wrapper::try_from(h.as_bytes())
+    }
+}
+
+/// [`Felt252Wrapper`] to [`H256`].
+impl From<Felt252Wrapper> for H256 {
+    fn from(felt: Felt252Wrapper) -> Self {
+        let buf: [u8; 32] = felt.into();
+        H256::from_slice(&buf)
+    }
+}
+
+/// [`Felt252Wrapper`] from [`FieldElement`].
+impl From<FieldElement> for Felt252Wrapper {
+    fn from(ff: FieldElement) -> Self {
+        Self(ff)
+    }
+}
+
+/// [`Felt252Wrapper`] to [`FieldElement`].
+impl From<Felt252Wrapper> for FieldElement {
+    fn from(ff: Felt252Wrapper) -> Self {
+        ff.0
+    }
+}
+
+/// [`Felt252Wrapper`] from [`Felt252`].
+impl From<Felt252> for Felt252Wrapper {
+    fn from(value: Felt252) -> Self {
+        Felt252Wrapper::try_from(&value.to_be_bytes()).unwrap()
+    }
+}
+
+/// [`Felt252Wrapper`] to [`Felt252`].
+impl From<Felt252Wrapper> for Felt252 {
+    fn from(felt: Felt252Wrapper) -> Self {
+        let buf: [u8; 32] = felt.into();
+        Felt252::from_bytes_be(&buf)
+    }
+}
+
+/// [`Felt252Wrapper`] from [`StarkFelt`].
+impl From<StarkFelt> for Felt252Wrapper {
+    fn from(value: StarkFelt) -> Self {
+        Felt252Wrapper::try_from(value.bytes()).unwrap()
+    }
+}
+
+/// [`Felt252Wrapper`] to [`StarkFelt`].
+impl From<Felt252Wrapper> for StarkFelt {
+    fn from(felt: Felt252Wrapper) -> Self {
+        let buf: [u8; 32] = felt.into();
+        StarkFelt::new(buf).unwrap()
+    }
+}
+
+/// SCALE trait.
+impl Encode for Felt252Wrapper {
+    fn encode_to<T: Output + ?Sized>(&self, dest: &mut T) {
+        dest.write(&self.0.to_bytes_be());
+    }
+}
```

Comment:
https://docs.substrate.io/reference/scale-codec/

I'm not sure at all that this is a good implementation.
`to_bytes_be` will return [u8; 32], 256 bits, even if the value is 0. With scale codec zero should usually be encoded as `0x00`. 
The whole purpose of scale codec is to reduce the size of the data being stored/send

Reply:
Also, scale codec is little-endian. When `to_bytes_be` is big-endian.
Scale is for: Simple Concatenated Aggregate Little-Endian.